types/sade/index.d.ts

```diff
@@ -47,7 +47,7 @@ declare namespace sade {
         help(str: string): void;
 
         parse(arr: string[], opts: { lazy: true } & ParseOptions): LazyOutput;
-        parse(arr: string[], opts?: ParseOptions): void;
+        parse(arr: string[], opts?: ParseOptions): any | Promise<any>;
```

Comment:
Why do you think this is a correct return?
There is no notion of promise/async in that source code,
I can see it is supposed to work that way:
```ts
Promise.resolve(prog.parse(process.argv)).catch(err => {
    if (!err.command) {
      console.error(err)
    } else if (err.stderr) {
      console.error(err.stderr)
    }
    process.exit(1)
  })
```
https://github.com/lukeed/sade/pull/31

Reply:
Then let's close it. I thought it was correct, since:

- I can supply an async function as action handler
- The return value of the handler is indeed returned from the parse call
- It does work as expected.

If it wasn't intended, this is obsolete then.

Reply:
Thx!